sdk/python/kfp/cli/recurring_run.py

```diff
@@ -0,0 +1,213 @@
+
+
+@click.group()
+def recurring_run():
+    """Manage recurring-run resources"""
+    pass
+
+
+@recurring_run.command()
+@click.option(
+    '--catchup/--no-catchup',
+    help='Whether the recurring run should catch up if behind schedule.',
+    type=bool)
+@click.option(
+    '--cron-expression',
+    help='A cron expression representing a set of times, using 6 space-separated fields, e.g. "0 0 9 ? * 2-6".'
+)
+@click.option('--description', help='The description of the recurring run.')
+@click.option(
+    '--enable-caching/--disable-caching',
+    help='Optional. Whether or not to enable caching for the run.',
+    type=bool)
+@click.option(
+    '--enabled/--disabled',
+    help='A bool indicating whether the recurring run is enabled or disabled.',
+    type=bool)
+@click.option(
+    '--end-time',
+    help='The RFC3339 time string of the time when to end the job.')
+@click.option(
+    '--experiment-id',
+    help='The ID of the experiment to create the recurring run under.')
+@click.option(
+    '--experiment-name',
+    help='The name of the experiment to create the recurring run under.')
+@click.option('--job-name', help='The name of the recurring run.')
+@click.option(
+    '--interval-second',
+    help='Integer indicating the seconds between two recurring runs in for a periodic schedule.'
+)
+@click.option(
+    '--max-concurrency',
+    help='Integer indicating how many jobs can be run in parallel.',
+    type=int)
+@click.option(
+    '--pipeline-id',
+    help='The ID of the pipeline to use to create the recurring run.')
+@click.option(
+    '--pipeline-package-path',
+    help='Local path of the pipeline package(the filename should end with one of the following .tar.gz, .tgz, .zip, .yaml, .yml).'
+)
+@click.option(
+    '--start-time',
+    help='The RFC3339 time string of the time when to start the job.')
+@click.option('--version-id', help='The id of a pipeline version.')
+@click.argument("args", nargs=-1)
+@click.pass_context
+def create(ctx: click.Context,
+           job_name: str,
+           experiment_id: Optional[str] = None,
+           experiment_name: Optional[str] = None,
+           catchup: Optional[bool] = None,
+           cron_expression: Optional[str] = None,
+           enabled: Optional[bool] = None,
+           description: Optional[str] = None,
+           enable_caching: Optional[bool] = None,
+           end_time: Optional[str] = None,
+           interval_second: Optional[int] = None,
+           max_concurrency: Optional[int] = None,
+           params: Optional[dict] = None,
+           pipeline_package_path: Optional[str] = None,
+           pipeline_id: Optional[str] = None,
+           start_time: Optional[str] = None,
+           version_id: Optional[str] = None,
+           args: Optional[List[str]] = None):
+    """Create a recurring run."""
+    client = ctx.obj["client"]
+    output_format = ctx.obj['output']
+
+    if experiment_id is None and experiment_name is None:
```

Comment:
How about `if (experiment_id is None) == (experiment_name is None):`? That that, we enforce one and only one of them must be provided. WDYT?

Reply:
Sounds great to me, learnt something new with that expression. Changed in the next commit!